### PR TITLE
Add new validators to validate input value is Alphanumeric and Email.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Add a new validator `AlphanumericValidator` to validate whether a given input is alphanumeric. 
Add a new validator `EmailFormatValidator` to validate whether a given input is in email format.

### CODE REVIEW

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/utils/Constants.java [hunk 1]
```diff
@@ -131,6 +133,10 @@ public enum ErrorMessages {
         ERROR_CODE_REGEX_MISMATCH("60021",
                 "Regex mismatch",
                 "The %s should satisfy the %s pattern."),
+        ERROR_VALIDATION_ALPHANUMERIC_MISMATCH("60022",
+                "ALPHANUMERIC_NOT_SATISFIED",
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/utils/Constants.java]: Addressed with commit 267e3b375990bd0e5aa3b06cf9f11ee9bb34d3a8

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.IS_ALPHANUMERIC;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_VALIDATION_ALPHANUMERIC_MISMATCH;
+
+/**
+ * Alphanumeric validator.
+ */
+public class AlphanumericValidator extends AbstractRulesValidator {
+
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:28]: Addressed

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+        String value = context.getValue();
+        String field = context.getField();
+        Map<String, String> attributesMap = context.getProperties();
+        String alphanumericRegEx = DEFAULT_ALPHANUMERIC_REGEX_PATTERN;
+
+        if (attributesMap.containsKey(IS_ALPHANUMERIC)) {
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:36]: Addressed

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            boolean isAlphanumeric = Boolean.parseBoolean(attributesMap.get(IS_ALPHANUMERIC));
+            if (isAlphanumeric && !value.matches(alphanumericRegEx)) {
+                throw new InputValidationMgtClientException(ERROR_VALIDATION_ALPHANUMERIC_MISMATCH.getCode(),
+                    ERROR_VALIDATION_ALPHANUMERIC_MISMATCH.getMessage(),
+                    String.format(ERROR_VALIDATION_ALPHANUMERIC_MISMATCH.getDescription(), field, alphanumericRegEx));
+            } else if (!isAlphanumeric && value.matches(alphanumericRegEx)) {
+                throw new InputValidationMgtClientException(ERROR_VALIDATION_ALPHANUMERIC_MISMATCH.getCode(),
+                    ERROR_VALIDATION_ALPHANUMERIC_MISMATCH.getMessage(),
+                    String.format(ERROR_VALIDATION_ALPHANUMERIC_MISMATCH.getDescription(), field, alphanumericRegEx));
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:41]: Addressed (removed else if block, as no need to do any validations if `isAlphanumeric` property is set to false.

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Alphanumeric validator.
+ */
+public class AlphanumericValidator extends AbstractRulesValidator {
+
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        String value = context.getValue();
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:30]: Addressed

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
+package org.wso2.carbon.identity.input.validation.mgt.model.validators;
```
Comment (kayathiri4) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java:1]: Add license header

Comment (kayathiri4) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java:1]: Check other files as well.

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+import java.util.Map;
+
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.DEFAULT_EMAIL_REGEX_PATTERN;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.IS_EMAIL;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_VALIDATION_EMAIL_MISMATCH;
+
+/**
+ * Email format validator.
+ */
+public class EmailValidator extends AbstractRulesValidator {
```
Comment (Thumimku) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java:18]:
```suggestion
public class EmailFormatValidator extends AbstractRulesValidator {
```
Email validator means something validate email not email name. WDYT?

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
+package org.wso2.carbon.identity.input.validation.mgt.model.validators;
```
Comment (Thumimku) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:1]: License

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java (hunk 6 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java]: Addressed with commit 0b70de75e94c399019b6a3ece166d73f95fcf31d

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java (hunk 7 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailValidator.java]: Addressed with commit 0b70de75e94c399019b6a3ece166d73f95fcf31d

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 8 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:19]: Addressed with commit 0b70de75e94c399019b6a3ece166d73f95fcf31d

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
```
Comment (kayathiri4) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java [hunk 10]
```diff
@@ -126,6 +126,20 @@ protected boolean validatePositiveNumber(String value, String property, String t
         return Integer.parseInt(value) < 0;
     }
 
+    protected boolean validateBoolean(String value, String property, String tenantDomain)
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java:129]: Can't we maintain the type of the value as a Boolean ?

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        String value = context.getValue();
+        String field = context.getField();
+        Map<String, String> attributesMap = context.getProperties();
+        String alphanumericRegEx = DEFAULT_ALPHANUMERIC_REGEX_PATTERN;
+
+        // Check whether value satisfies the alphanumeric criteria.
+        if (attributesMap.containsKey(ENABLE)) {
+            if (Boolean.parseBoolean(attributesMap.get(ENABLE)) && value != null && !value.matches(alphanumericRegEx)) {
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:66]: Use Stringutils to check null or empty the string value != null > StringUtils.isBlank(value)

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+                throw new InputValidationMgtClientException(ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH.getCode(),
+                    ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH.getMessage(), String.format(
+                    ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH.getDescription(), field, alphanumericRegEx));
+            }
+        }
+
+        return true;
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:73]: return Boolean.TRUE

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+     *
+     * @return  List<Property>
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+        int parameterCount = 0;
+
+        Property enable = new Property();
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:87]: Property alphaNumericValidator = new Property();
alphaNumericValidator. setName (ENABLE_VALIDATOR)

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java [hunk 14]
```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java:2]: 2023

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+     *
+     * @return  List<Property>
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+        int parameterCount = 0;
+
+        Property enable = new Property();
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java:87]: Modify the name

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+        // Check whether value satisfies the email format criteria.
+        if (attributesMap.containsKey(ENABLE)) {
+            if (Boolean.parseBoolean(attributesMap.get(ENABLE)) && value != null && !value.matches(emailRegEx)) {
+                throw new InputValidationMgtClientException(ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH.getCode(),
+                        ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH.getMessage(),
+                        String.format(ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH.getDescription(), field, emailRegEx));
+            }
+        }
+
+        return true;
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java:73]: Boolean.true

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 11 above)
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:66]: Is it possible to become the 'value' null or attributes map does not contain the 'enabled' property? If so we need to throw correct errors. In the logic that part is not handled.

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java [hunk 17]
```diff
@@ -126,6 +126,20 @@ protected boolean validatePositiveNumber(String value, String property, String t
...
+
+        if (!Boolean.FALSE.toString().equalsIgnoreCase(value) && !Boolean.TRUE.toString().equalsIgnoreCase(value)) {
+            if (log.isDebugEnabled()) {
+                log.error(String.format("The property %s should be a %s value for the tenant %s.", property,
+                        "boolean", tenantDomain));
+            }
+            throw new InputValidationMgtClientException(ERROR_PROPERTY_TYPE_MISMATCH.getCode(),
+                String.format(ERROR_PROPERTY_TYPE_MISMATCH.getDescription(), property, "boolean", tenantDomain));
+        }
+        return true;
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java:140]: Boolean.TRUE

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+     * @return  boolean
+     */
+    @Override
+    public boolean validateProps(ValidationContext context) throws InputValidationMgtClientException {
+
+        Map<String, String> properties = context.getProperties();
+        validatePropertyName(properties, this.getClass().getSimpleName(), context.getTenantDomain());
+        if (properties.get(ENABLE) != null && !validateBoolean(properties.get(ENABLE),
+                ENABLE, context.getTenantDomain())) {
+            properties.remove(ENABLE);
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java:112]: This should be an error scenario

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        String value = context.getValue();
+        String field = context.getField();
+        Map<String, String> attributesMap = context.getProperties();
+        String emailRegEx = DEFAULT_EMAIL_REGEX_PATTERN;
+
+        // Check whether value satisfies the email format criteria.
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java:64]: Thorw error if value == null

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+     * @return  boolean
+     */
+    @Override
+    public boolean validateProps(ValidationContext context) throws InputValidationMgtClientException {
+
+        Map<String, String> properties = context.getProperties();
+        validatePropertyName(properties, this.getClass().getSimpleName(), context.getTenantDomain());
+        if (properties.get(ENABLE) != null && !validateBoolean(properties.get(ENABLE),
+                ENABLE, context.getTenantDomain())) {
+            properties.remove(ENABLE);
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java:111]: Throw an error

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 9 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java (hunk 10 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java:132]: No, As per swagger definition, property values are always strings

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 11 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 13 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java (hunk 15 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 11 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java (hunk 18 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java (hunk 19 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java:67]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java (hunk 20 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java (hunk 14 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java]: Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2

**Review by Thumimku — APPROVED** [review]
Apart from adding additional validators, you changed some behaviors like throwing errors.
Please explain the modified logic in the Issue description.
Please run tests before merging PR
LGTM

**Pull request merged**